qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -520,6 +523,82 @@ def _check_special_columns(cls, md_template, obj):
             cls._check_template_special_columns(md_template, obj))
 
     @classmethod
+    def _clean_validate_template(cls, md_template, study_id, obj,
+                                 conn_handler=None):
+        """Takes care of all validation and cleaning of metadata templates
+
+        Parameters
+        ----------
+        md_template : DataFrame
+            The metadata template file contents indexed by sample ids
+        study_id : int
+            The study to which the metadata template belongs to.
+        obj : object
+            Any extra object needed by the template to perform any extra check
+
+        Returns
+        -------
+        md_template : DataFrame
+            Cleaned copy of the input md_template
+
+        Raises
+        ------
+        QiitaDBColumnError
+            If the sample names in md_template contains invalid names
+        QiitaDBDuplicateHeaderError
+            If md_template contains duplicate headers
+        QiitaDBColumnError
+            If md_template is missing a required column
+        """
+        cls._check_subclass()
+        invalid_ids = get_invalid_sample_names(md_template.index)
+        if invalid_ids:
+            raise QiitaDBColumnError("The following sample names in the sample"
+                                     " template contain invalid characters "
+                                     "(only alphanumeric characters or periods"
+                                     " are allowed): %s." %
+                                     ", ".join(invalid_ids))
+        # We are going to modify the md_template. We create a copy so
+        # we don't modify the user one
+        md_template = deepcopy(md_template)
+
+        # Prefix the sample names with the study_id
+        prefix_sample_names_with_id(md_template, study_id)
+
+        # In the database, all the column headers are lowercase
+        md_template.columns = [c.lower() for c in md_template.columns]
+
+        # Check that we don't have duplicate columns
+        if len(set(md_template.columns)) != len(md_template.columns):
+            raise QiitaDBDuplicateHeaderError(
```

Comment:
Hey, just noticed that this error message is not going to be very information (and it is not your fault, the code was like this already). If possible can you change to something along the lines of: `'The following header names are duplicated: %s' % ', '.join(find_duplicates(md_template.columns))`